x/incentives/keeper/gauge_test.go

```diff
@@ -280,33 +114,92 @@ func (suite *KeeperTestSuite) TestGaugesByDenom() {
 		err = suite.App.IncentivesKeeper.MoveUpcomingGaugeToActiveGauge(suite.Ctx, *gauge)
 		suite.Require().NoError(err)
 
+		// check upcoming gauges
+		gauges = suite.App.IncentivesKeeper.GetUpcomingGauges(suite.Ctx)
+		suite.Require().Len(gauges, 0)
+
 		// check gauge ids by denom
-		gaugeIds = suite.App.IncentivesKeeper.GetAllGaugeIDsByDenom(suite.Ctx, "lptoken")
-		suite.Require().Len(gaugeIds, 1)
+		gaugeIdsByDenom = suite.App.IncentivesKeeper.GetAllGaugeIDsByDenom(suite.Ctx, "lptoken")
+		suite.Require().Len(gaugeIdsByDenom, 1)
 
 		// check gauge ids by other denom
-		gaugeIds = suite.App.IncentivesKeeper.GetAllGaugeIDsByDenom(suite.Ctx, "lpt")
-		suite.Require().Len(gaugeIds, 0)
+		gaugeIdsByDenom = suite.App.IncentivesKeeper.GetAllGaugeIDsByDenom(suite.Ctx, "lpt")
+		suite.Require().Len(gaugeIdsByDenom, 0)
 
 		// distribute coins to stakers
-		_, err = suite.App.IncentivesKeeper.Distribute(suite.Ctx, []types.Gauge{*gauge})
+		distrCoins, err := suite.App.IncentivesKeeper.Distribute(suite.Ctx, []types.Gauge{*gauge})
 		suite.Require().NoError(err)
+		suite.Require().Equal(sdk.Coins{sdk.NewInt64Coin("stake", int64(10/tc.expectedNumEpochsPaidOver))}, distrCoins)
+
+		if tc.isPerpetual {
+			// distributing twice without adding more for perpetual gauge
+			gauge, err = suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, gaugeID)
+			suite.Require().NoError(err)
+			distrCoins, err = suite.App.IncentivesKeeper.Distribute(suite.Ctx, []types.Gauge{*gauge})
+			suite.Require().NoError(err)
+			suite.Require().True(distrCoins.Empty())
+
+			// add to gauge
+			addCoins := sdk.Coins{sdk.NewInt64Coin("stake", 200)}
+			suite.AddToGauge(addCoins, gaugeID)
+
+			// distributing twice with adding more for perpetual gauge
+			gauge, err = suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, gaugeID)
+			suite.Require().NoError(err)
+			distrCoins, err = suite.App.IncentivesKeeper.Distribute(suite.Ctx, []types.Gauge{*gauge})
+			suite.Require().NoError(err)
+			suite.Require().Equal(sdk.Coins{sdk.NewInt64Coin("stake", 200)}, distrCoins)
+		}
+
+		// check active gauges
+		gauges = suite.App.IncentivesKeeper.GetActiveGauges(suite.Ctx)
+		suite.Require().Len(gauges, 1)
+
+		// check gauge ids by denom
+		gaugeIdsByDenom = suite.App.IncentivesKeeper.GetAllGaugeIDsByDenom(suite.Ctx, "lptoken")
+		suite.Require().Len(gaugeIdsByDenom, 1)
 
 		// finish distribution for non perpetual gauge
```

Comment:
This whole section should be moved under the non-perpetual gauge check on line 169 since it has the same condition right?